### ISSUE TITLE
Size of 0install installation window.

### ISSUE THREAD

**Issue by ReneSac** [0install/0install#33]
Oinstall installation window (where it first shows the dependencies) is wider than my screen by default (1080 wide, a portrait monitor). Even if I resize it, in the next package I install it returns to the same size as before. I can also move it to my other wider screen, but the next time it opens it will aways open in the narrower screen, regardless of where the main 0install applications window is. It is annoying.

**Comment by talex5** [comment]
What version of 0install and what OS are you using?

**Comment by ReneSac** [comment]
Sorry. I'm using Kubuntu 14.10, and the 0install version is 2.7 (the one from apt-get).

**Comment by talex5** [comment]
The code that sets it is here:

https://github.com/0install/0install/blob/d9c8219d603176a77d2650322f7054463b1c58c6/ocaml/gui_gtk/solver_box.ml#L135

    dialog#set_default_size
      ~width:(Gdk.Screen.width () * 2 / 5)
      ~height:300;

So, it should be 2/5 of the width of the screen, but perhaps it's the total width of both screens?

**Comment by ReneSac** [comment]
Yeah, from the documentation: "a screen may consist of multiple monitors that are merged to form a large screen area".

The correct way seems to be a combination of `gtk.gdk.Screen.get_monitor_at_point` or `gtk.gdk.Screen.get_monitor_at_window` and `gtk.gdk.Screen.get_monitor_geometry`. It seems that there is no equivalent of `gtk.gdk.Screen.get_height` in pixels for a monitor, only in mm... I don't know that is available in the Ocalm bindings.

**Comment by talex5** [comment]
Yes, not obvious how to do it. Perhaps we could assume a square screen of size `min width height` and make it some fraction of that. I use a tiling WM, so initial window sizes are ignored for me.
